src/EditorFeatures/Core/Implementation/IntelliSense/Completion/CompletionProviders/AbstractPartialCompletionProvider.cs

```diff
@@ -39,18 +39,28 @@ public AbstractPartialCompletionProvider(IWaitIndicator waitIndicator)
         protected abstract Task<TextSpan> GetTextChangeSpanAsync(Document document, int position, CancellationToken cancellationToken);
         protected abstract bool IsPartial(IMethodSymbol m);
 
-        protected override async Task<IEnumerable<CompletionItem>> GetItemsWorkerAsync(Document document, int position, CompletionTriggerInfo triggerInfo, CancellationToken cancellationToken)
+        public override async Task ProduceCompletionListAsync(CompletionListContext context)
         {
+            var document = context.Document;
+            var position = context.Position;
+            var cancellationToken = context.CancellationToken;
+
```

Comment:
After double-checking, each of these variables is used twice below so I'll go ahead and leave the locals.